blog/03-07-2023-escrow-service-on-celo-that-holds-funds-until-certain-conditions-are-met/escrow-service-on-celo-that-holds-funds-until-certain-conditions-are-met.md

```diff
@@ -0,0 +1,448 @@
+                }
+                else if(_user == seller && sellerBalance != 0) {
+                    fee = sellerBalance.mul(transferFee).div(100);
+                    seller.transfer(sellerBalance.sub(fee));
+                    feeAddress.transfer(fee);
+                    sellerBalance = 0;
+                }
+        }
+  
+    }
+
+```
+
+9. Return functions
+
+This smart contract contains a series of functions that allow other contracts or external parties to access specific data stored in the contract. Each function is defined with the public keyword, which means that they can be accessed by anyone on the network. The view keyword indicates that these functions do not modify any data within the contract and simply return data.
+
+Here's a brief explanation of each function:
+
+- returnSeller() - This function returns the address of the seller in the contract.
+
+    ```solidity
+    function returnSeller() public view returns(address){
+        return seller;
+    }
+    ```
+
+- returnBuyer() - This function returns the address of the buyer in the contract.
+
+    ```solidity
+    function returnBuyer() public view returns(address){
+        return buyer;
+    }
+    ```
+
+- returnSellerBalance() - This function returns the current balance of the seller in the contract.
+
+    ```solidity
+    function returnSellerBalance() public view returns(uint){
+        return sellerBalance;
+    }
+    ```
+
+- returnBuyerBalance() - This function returns the current balance of the buyer in the contract.
+
+    ```solidity
+    function returnBuyerBalance() public view returns(uint){
+        return buyerBalance;
+    }
+    ```
+
+- returnSecurityDeposit() - This function returns the security deposit amount stored in the contract.
+
+    ```solidity
+    function returnSecurityDeposit() public view returns(uint){
+        return securityDeposit;
+    }
+    ```
+
+- returnSellerPreAgreementAmount() - This function returns the amount that the seller had agreed to before the contract was created.
+
+    ```solidity
+    function returnSellerPreAgreementAmount() public view returns(uint){
+        return sellerPreAgreementAmount;
+    }
+    ```
+
+- returnBuyerPreAgreementAmount() - This function returns the amount that the buyer had agreed to before the contract was created.
+
+    ```solidity
+    function returnBuyerPreAgreementAmount() public view returns(uint){
+        return buyerPreAgreementAmount;
+    }
+    ```
+
+- returnEndDate() - This function returns the end date of the contract.
+
+    ```solidity
+    function returnEndDate() public view returns(uint){
+        return endDate;
+    }
+    ```
+
+- returnDaysForWithdrawal() - This function returns the number of days allowed for withdrawal after the contract is created.
+
+    ```solidity
+    function returnDaysForWithdrawal() public view returns(uint){
+        return daysForWithdrawal;
+    }
+    ```
+
+## Full Code
+
+    ```solidity
```

Comment:
I'm guessing the tab here was an accident 
remove the tab before the "```solidity", it's altering your code snippet from rendering properly

Reply:
done